ui/src/views/Tasks/TaskGroup/index.jsx

```diff
@@ -174,6 +175,7 @@ export default class TaskGroup extends Component {
 
     if (prevProps.match.params.taskGroupId !== taskGroupId) {
       updateTaskGroupIdHistory(taskGroupId);
+      this.subscribeToMore();
```

Comment:
There's a bit of a race condition here, since the "current" state is loaded, then the pulse listeners are established.  If the state of a task changes in that interval, the UI will never show it.  I can't really tell if this is a big deal or not, though -- it depends on the propagation delay of pulse messages and how much time elapses between fetching the base task state and beginning the subscription.

We have the advantage that applying pulse messages redundantly still yields a correct result.  Is it possible to switch the order, so that the subscription begins *before* the current state is fetched?

Reply:
Good observation. I'll look into it.

Reply:
According to [react docs](https://reactjs.org/docs/react-component.html#mounting), the order in which the methods are called when mounting is:

1. constructor()
2. static getDerivedStateFromProps()
3. render()
4. componentDidMount()

I added a commit that moves the listener creation from `componentDidMount` to `getDerivedStateFromProps`. This should help in making it less prone to a race condition.